### PR TITLE
feat(sheets-data-validation): optimize data-validation error msg

### PR DESCRIPTION
**Pull request by weird94**
## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/data-validation/src/validators/base-data-validator.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ export abstract class BaseDataValidator<DataType = CellValue> {
         return `${this.titleStr} ${ruleName}`;
     }
 
-    generateRuleErrorMessage(rule: IDataValidationRuleBase) {
+    generateRuleErrorMessage(rule: IDataValidationRuleBase, position: any) {
```
Comment (wzhudev) [inline packages/data-validation/src/validators/base-data-validator.ts:102]: 不要用 `any`

File: packages/sheets-data-validation/src/validators/text-length-validator.ts [hunk 2]
```diff
@@ -14,20 +14,22 @@
...
-
-const FORMULA1 = '{FORMULA1}';
-const FORMULA2 = '{FORMULA2}';
+import { FORMULA1, FORMULA2 } from './const';
+import { getTransformedFormula } from './util';
 
 export class TextLengthValidator extends BaseDataValidator<number> {
     id: string = DataValidationType.TEXT_LENGTH;
     title: string = 'dataValidation.textLength.title';
+    private _lexerTreeBuilder = this.injector.get(LexerTreeBuilder);
```
Comment (wzhudev) [inline packages/sheets-data-validation/src/validators/text-length-validator.ts:32]: rugo
```suggestion
    private readonly _lexerTreeBuilder = this.injector.get(LexerTreeBuilder);
```

File: packages/sheets-data-validation/src/validators/text-length-validator.ts [hunk 3]
```diff
@@ -195,12 +197,13 @@ export class TextLengthValidator extends BaseDataValidator<number> {
         return cellValue <= formula1;
     }
 
-    override generateRuleErrorMessage(rule: IDataValidationRuleBase) {
+    override generateRuleErrorMessage(rule: IDataValidationRuleBase, pos: ISheetLocationBase) {
```
Comment (wzhudev) [inline packages/sheets-data-validation/src/validators/text-length-validator.ts:200]: 这里就是有类型的，上面应该也可以不用 `any` 呢？

**Review by wzhudev — DISMISSED** [review]
Some code readability issue need to be addressed.

**Review by Gggpound — APPROVED** [review]

**Pull request merged**